### PR TITLE
Generate Slurm config files from templates

### PR DESCRIPTION
**Pull request by nathanielknight**
This commit updates the `slurm_node` Ansible role to generate the
`slurm.conf` file from a template. This lets different environments
specify different numbers, and names of hosts while using the same
roles.

Currently, only the name and RAM of Slurm hosts is configurable, but
adding other fields to the template will be straightforward if it
becomes necessary.

### CODE REVIEW

File: dev-env/dev_env_vars.yml [hunk 1]
```diff
@@ -23,5 +23,9 @@ kive_httpd_user: kive
 kive_httpd_group: kive
 # - package variables
 slurmbuilddir: "/root"
+slurmnodes:
+    - name: head
+      memory: "7000"
```
Comment (donkirkby) [inline dev-env/dev_env_vars.yml:28]: Please add a comment with the memory units, presumably MB.

Comment (nathanielknight) [reply dev-env/dev_env_vars.yml:28]: This just gets substituted into the config file template, so the units are the same as what Slurm takes (presumable, yes, MB).

File: dev-env/ssh-config [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Host default
+  HostName 127.0.0.1
+  User vagrant
+  Port 2222
+  UserKnownHostsFile /dev/null
+  StrictHostKeyChecking no
+  PasswordAuthentication no
+  IdentityFile C:/Users/natha/work/Kive/dev-env/.vagrant/machines/default/virtualbox/private_key
```
Comment (donkirkby) [inline dev-env/ssh-config:8]: Did you mean to commit this file? This line particularly looks like it's specific to your workstation.

Comment (nathanielknight) [reply dev-env/ssh-config]: 🤦‍♂️ Good catch, thanks.

**Pull request merged**